src/providers/details/tabs/Hosts/components/VSphereNetworkModal.tsx

```diff
@@ -0,0 +1,103 @@
+import { type FC, type ReactNode, useCallback, useState } from 'react';
+import { AlertMessageForModals } from 'src/modules/Providers/modals/components/AlertMessageForModals';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import ModalForm from '@components/ModalForm/ModalForm';
+import type { NetworkAdapters, V1beta1Provider } from '@kubev2v/types';
+import { Form, ModalVariant } from '@patternfly/react-core';
+
+import { getSelectedInventoryHostNetworkTriples } from './utils/getSelectedInventoryHostNetworkTriples';
+import { onSaveHost } from './utils/onSaveHost';
+import type { InventoryHostNetworkTriple } from './utils/types';
+import { validatePassword, validateUsername } from './utils/validators';
+import HostsNetworksSelect from './HostsNetworksSelect';
+import HostsNetworksSetPassword from './HostsNetworksSetPassword';
+import HostsNetworksSetUserName from './HostsNetworksSetUserName';
+
+import './VSphereNetworkModal.style.css';
+
+type VSphereNetworkModalProps = {
+  provider: V1beta1Provider;
+  data: InventoryHostNetworkTriple[];
+  selectedIds: string[];
+};
+
+const VSphereNetworkModal: FC<VSphereNetworkModalProps> = ({ data, provider, selectedIds }) => {
+  const { t } = useForkliftTranslation();
+  const [alertMessage, setAlertMessage] = useState<ReactNode>(null);
+  const [network, setNetwork] = useState<NetworkAdapters>(undefined as unknown as NetworkAdapters);
+  const [username, setUsername] = useState<string>(undefined as unknown as string);
+  const [password, setPassword] = useState<string>(undefined as unknown as string);
+
+  const selectedLength = selectedIds.length;
+  const endpointType = provider?.spec?.settings?.sdkEndpoint ?? '';
+
+  const shouldDisableSave = () => {
+    if (endpointType === 'esxi') {
+      return network === undefined;
+    }
+
+    return (
+      network === undefined ||
+      !username ||
+      !password ||
+      !validateUsername(username) ||
+      !validatePassword(password)
+    );
+  };
+
+  const selectedInventoryHostPairs = getSelectedInventoryHostNetworkTriples(data, selectedIds);
+
+  const handleSave = useCallback(async () => {
+    try {
+      await onSaveHost({
+        hostPairs: selectedInventoryHostPairs,
+        network,
+        passwd: endpointType === 'esxi' ? undefined : password,
+        provider,
+        user: endpointType === 'esxi' ? undefined : username,
+      });
+    } catch (err) {
+      setAlertMessage(<AlertMessageForModals title={t('Error')} message={err?.toString()} />);
```

Comment:
Sharon, we are trying to avoid past mistakes, but saving a component in a state isn't a good practice.
As we discussed in very early PRs, please don't save the component in the state, only the changing string of the message.

Reply:
I removed this whole catch block since there is no need for that, it will be anyway catched and handled in ModalForm